### PR TITLE
Issue 739/hazardous module loading

### PR DESCRIPTION
**Pull request by cans**
This is a fix for #739. ~~it is based on #740, consider only the last two commits~~

A few assertions were added to find potential issues. They may be removed before merging.

### CODE REVIEW

**Comment by webknjaz** [comment]
This can be rebased now

File: test/TestAbsPath.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8; -*-
+import os
+
+from ansiblelint.cli import abspath
+
+
+def test_expand_path_vars(monkeypatch, tmpdir):
+    test_path = os.path.join(str(tmpdir), 'path')  # str(tmpdir): Issue with Py3.5
```
Comment (webknjaz) [inline test/TestAbsPath.py:8]: I believe we figured out earlier that using the `tmp_path` fixture should be preferred over `tmpdir`.

**Comment by webknjaz** [comment]
@cans FYI I cherry-picked https://github.com/ansible/ansible-lint/pull/741/commits/d94fa9a8078de3b794ed41d530631854846c9d6b into master

**Comment by awcrosby** [comment]
@cans this can be rebased with just the last 2 commits, that should resolve the conflict

**Comment by ssbarnea** [comment]
@cans Can you please convert this to Draft (top-right corner)?

**Comment by webknjaz** [comment]
@cans FYI we've decided to drop support for Python 3.5

**Comment by ssbarnea** [comment]
Closing as we only rely on Ansible behavior, nothing specific to the linter.

**Pull request closed without merging**